### ISSUE TITLE
jupyter serverextension enable fails, can not import nbgrader.server_extensions.assignment_list 

### ISSUE THREAD

**Issue by mboldin-temple** [jupyter/nbgrader#1334]
### Operating system
Windows 10   32 bit
### `nbgrader --version`
6.1
### `jupyter notebook --version` 
6.0.0
Python 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 19:29:22) [MSC v.1916 32 bit (Intel)]
### Expected behavior
jupyter serverextension enable --sys-prefix --py nbgrader
add FORMGRADER and other items to Jupyter  
### Actual behavior
running 
jupyter serverextension enable --sys-prefix --py nbgrader
produces
Enabling: nbgrader.server_extensions.assignment_list
- Writing config: c:\users\mboldin\appdata\local\programs\python\python37-32\etc\jupyter
    - Validating...
Error loading server extension nbgrader.server_extensions.assignment_list
      X is nbgrader.server_extensions.assignment_list importable?

### Steps to reproduce the behavior
pip install nbgrader
jupyter nbextension install --sys-prefix --py nbgrader --overwrite
jupyter nbextension enable --sys-prefix --py nbgrader
jupyter serverextension enable --sys-prefix --py nbgrader

also 
>>> import nbgrader
>>> import nbgrader.server_extensions.assignment_list

  File "C:\Users\mboldin\AppData\Local\Programs\Python\Python37-32\lib\site-packages\nbgrader\api.py", line 994, in <module>
    Grade.needs_manual_grade))
AttributeError: '_Exists' object has no attribute 'correlate_except'

**Comment by willingc** [comment]
@mboldin-temple Thanks for the issue report.

If you run these two commands what is the output?
- `jupyter nbextension list`
- `jupyter serverextension list`

**Comment by willingc** [comment]
I would also go ahead and update your installation of the jupyter notebook to 6.0.3 with `pip install -U jupyter`

**Comment by mboldin-temple** [comment]
>jupyter nbextension list
Known nbextensions:
  config dir: C:\Users\mboldin\.jupyter\nbconfig
    notebook section
      latex_envs/latex_envs enabled
      - Validating: ok
  config dir: c:\users\mboldin\appdata\local\programs\python\python37-32\etc\jupyter\nbconfig
    notebook section
      jupyter-js-widgets/extension enabled
      - Validating: ok
      create_assignment/main enabled
      - Validating: ok
      validate_assignment/main enabled
      - Validating: ok
    tree section
      formgrader/main enabled
      - Validating: ok
      assignment_list/main enabled
      - Validating: ok

>jupyter serverextension list
config dir: C:\Users\mboldin\.jupyter
    nbgrader.server_extensions.formgrader enabled
    - Validating...
Error loading server extension nbgrader.server_extensions.formgrader
      X is nbgrader.server_extensions.formgrader importable?
    nbgrader.server_extensions.validate_assignment enabled
    - Validating...
Error loading server extension nbgrader.server_extensions.validate_assignment
      X is nbgrader.server_extensions.validate_assignment importable?
config dir: c:\users\mboldin\appdata\local\programs\python\python37-32\etc\jupyter
    nbgrader.server_extensions.formgrader enabled
    - Validating...
Error loading server extension nbgrader.server_extensions.formgrader
      X is nbgrader.server_extensions.formgrader importable?
    nbgrader.server_extensions.validate_assignment enabled
    - Validating...
Error loading server extension nbgrader.server_extensions.validate_assignment
      X is nbgrader.server_extensions.validate_assignment importable?
    nbgrader.server_extensions.assignment_list enabled
    - Validating...
Error loading server extension nbgrader.server_extensions.assignment_list
      X is nbgrader.server_extensions.assignment_list importable?

**Comment by willingc** [comment]
Hmm...it looks like the serverextension is not installed fully.

Please try to disable and then enable again:
```
testenv ❯ jupyter serverextension disable --py nbgrader

Disabling: nbgrader.server_extensions.formgrader
- Writing config: /Users/willingc/.jupyter
Disabling: nbgrader.server_extensions.validate_assignment
- Writing config: /Users/willingc/.jupyter
Disabling: nbgrader.server_extensions.assignment_list
- Writing config: /Users/willingc/.jupyter
Disabling: nbgrader.server_extensions.course_list
- Writing config: /Users/willingc/.jupyter
```

and then enable:
```
testenv ❯ jupyter serverextension enable --python nbgrader

Enabling: nbgrader.server_extensions.formgrader
- Writing config: /Users/willingc/.jupyter
    - Validating...
      nbgrader.server_extensions.formgrader  OK
Enabling: nbgrader.server_extensions.validate_assignment
- Writing config: /Users/willingc/.jupyter
    - Validating...
      nbgrader.server_extensions.validate_assignment  OK
Enabling: nbgrader.server_extensions.assignment_list
- Writing config: /Users/willingc/.jupyter
    - Validating...
      nbgrader.server_extensions.assignment_list  OK
Enabling: nbgrader.server_extensions.course_list
- Writing config: /Users/willingc/.jupyter
    - Validating...
      nbgrader.server_extensions.course_list  OK
```

Hopefully that will resolve the issue.

**Comment by mboldin-temple** [comment]
Still same p[roblem after disable and enable again

Error loading server extension nbgrader.server_extensions.assignment_list
      X is nbgrader.server_extensions.assignment_list importable?

Note tried to   import nbgrader.server_extensions.assignment_list in a
python script.
and this fails.

One possibility is a conflict from a second install of Python 64 bit to run
Jupyter as 64 bit app.
The fails are for my main 32 bit versions
The 64 bit version fails for other reasons -- problem figuring out the
proper directory

**Comment by mboldin-temple** [comment]
But uninstalling both jupyters and re-installing for Python 32 bit did
not fix the problem

**Comment by willingc** [comment]
@mboldin-temple Hmm... Does your system support 64 bit? If so, is there are particular use case which you are needing 32 bit?

> In Windows 10, go to Settings > System > About or type About in the Windows 10 search box. Under the Device specifications heading, you'll see it at System type: "x-bit operating system, x??-based processor". 

The above is a way to check if the processor can support 64 bit. 

If you need to use 32 bit, you may try using a virtual environment when running your python packages:
```
python3 -m venv my_test_env
source my_test_env/bin/activate
pip install jupyter
pip install nbgrader
jupyter nbextension install --sys-prefix --py nbgrader --overwrite
jupyter nbextension enable --sys-prefix --py nbgrader
jupyter serverextension enable --sys-prefix --py nbgrader
```

**Comment by mboldin-temple** [comment]
I figured it out and there was not a good reason I was runing 32 bit except this how I started  on a relatively new Windows laptop.    

By setting up a virtual environment, nbgrader full installed and the serverextensions worked. I then saw the virtualenv was 64 bit.  What I believe was happening  was that some steps in my nbgrader install and setup were mixing my 32 bit and 64 bit python paths because I was not consistent in the Windows Path and used both local user and system user configurations.   I will just re-install everything Python related to be v3.7 64-bit and get rid of the mixed versions. 

So thanks for the virtualenv suggestion.
